PDF fix and update

SecondSkoll opened the pull request:
- Fixes the error causing RTD and local generation to fail (see my comment in #58)
- Updates the PDF to use the current branding (still in beta - not in the starter pack yet - but approved by Design team)
  - Includes use of Ubuntu font family, modern branding, and some general structural fixes
- Adds Make entrypoints for local PDF generation (including option to force installation of required system packages)
  - `make pdf` will install and use VENV to generate the PDF and remove build directory, if system packages are missing context will be provided and process will exit. `make pdf-prep-force` installs apt packages required for PDF generation - this is separate as it's not containerised and affects the local system.
  
[This is the verified output](https://drive.google.com/file/d/1yfC_z8f3fVFP7WNuAU2odgczayXpZihm/view?usp=sharing). RTD generation should be 1:1, but will need to be checked upon merge.

SecondSkoll committed:
Add PNG version of CoF

SecondSkoll committed:
PDF fix and update

Fixes SVG issues by replacing with PNG
Updates branding to current standard
Adds entrypoint to Makefile for creating PDFs

SecondSkoll commented:
Converted to draft to address query - "Would it be possible to have a cover that would not waste a lot of ink if printed?".

SecondSkoll committed:
Change PDF title page to a light mode version

SecondSkoll commented:
I've gone ahead and changed the title page to a much more printer-friendly version - but I did just put it together myself, so I'll have to work with the Design team to get a proper printer-friendly title page approved. It will likely take some time (probable ETA next cycle), so I think this should do for now.

SecondSkoll commented:
Note: The broken links from the Documentation Checks are functioning - but manpages.ubuntu.com doesn't seem to like the linkchecker.

s-makin commented:
> I've gone ahead and changed the title page to a much more printer-friendly version - but I did just put it together myself, so I'll have to work with the Design team to get a proper printer-friendly title page approved. It will likely take some time (probable ETA next cycle), so I think this should do for now.

I *really* like the light version of the front page. It looks really good!

s-makin commented on docs/custom_conf.py:
```diff
@@ -262,3 +259,123 @@
+}
+\fancypagestyle{titlepage}{%
+    \fancyhf{}
+    \fancyfoot[L]{\footnotesize \textcolor{copyright}{© 2024 Canonical Ltd. All rights reserved. \newline Confidential and proprietary, do not share without permission.}}
```
As we did with the Server guide, we'll probably want to change this line as well. The Packaging Guide isn't confidential :)

dviererbe replied:
+1

suggestion: According to the [Software Licensing](https://sites.google.com/a/canonical.com/operations/legal/software-licensing?authuser=0&pli=1#h.w171o4ftougj) guide from the legal team we should probably put in something like this:

```suggestion
    \fancyfoot[L]{\footnotesize \textcolor{copyright}{© 2024 Canonical Ltd. This work is licensed under the Creative Commons Attribution-Share Alike 3.0 Unported License.}}
```

s-makin reviewed (commented):
Apart from one little nit, LGTM :) it'll be nice to have the local PDF build ability as well - although we might want to also add a little documentation to explain how to use it (or add it to the make "help" list so when users run ``make`` they can see it as an option)

dviererbe commented on docs/Makefile:
```diff
@@ -99,6 +99,35 @@ woke: woke-install
 pa11y: pa11y-install html
 	find $(BUILDDIR) -name *.html -print0 | xargs -n 1 -0 $(PA11Y)
 
+# Standard PDF generation checks for system packages and provides context if not found
+
+pdf-prep: install
+	@. $(VENV); (dpkg-query -W -f='$${Status}' latexmk 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package latexmk is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex plantuml xindy te ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' fonts-freefont-otf 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package fonts-freefont-otf is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-x ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' fonts-ibm-plex 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package fonts-ibm-plex is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex pla ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' texlive-latex-recommended 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package texlive-latex-recommended is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' texlive-latex-extra 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package texlive-latex-extra is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' texlive-fonts-recommended 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package texlive-fonts-recommended is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' texlive-font-utils 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package texlive-font-utils is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-x ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' texlive-lang-cjk 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package texlive-lang-cjk is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' texlive-xetex 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package texlive-xetex is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex plant ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' plantuml 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package plantuml is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex plantuml xindy  ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' xindy 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package xindy is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex plantuml xindy tex-gy ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' tex-gyre 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package tex-gyre is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex plantuml xindy  ...
+	@. $(VENV); (dpkg-query -W -f='$${Status}' dvipng 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package dvipng is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex plantuml xindy tex- ...
+
+# Entrypoint to install system packages, separate from usual workflow due to permanence
+
+pdf-prep-force:
+	@. $(VENV); apt-get update
+	@. $(VENV); apt-get upgrade -y
+	@. $(VENV); apt-get install --no-install-recommends -y latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex plantuml xindy tex-gyre dvipng \
+
+pdf: pdf-prep
+	@. $(VENV); sphinx-build -M latexpdf "$(SOURCEDIR)" "_build" $(SPHINXOPTS)
+	@. $(VENV); find ./_build/latex -name "*.pdf" -exec mv -t ./ {} +
+	@. $(VENV); rm -r _build
+
```
note: With all the dpkg and apt dependencies it will probably take forever to get this into the sphinx starter pack 😅, but that is luckily not my concern.

SecondSkoll replied:
I view this as optional, and I've implemented it this way to support local generation without sneakily installing packages. As other validation tools [do this](https://github.com/canonical/sphinx-docs-starter-pack/issues/244) it hopefully won't be too hard to get it included.

dviererbe replied:
Fine with me. This is Ubuntu documentation and does not need to work on any other platform.

dviererbe commented on docs/images/reference/glossary/CoF-Square.svg:
note: I really would have liked to keep the svg's, but it looks like there is no conditional in reStructuredText to use png's only for LaTeX aside from using the [`raw` instruction](https://docutils.sourceforge.io/docs/ref/rst/directives.html#raw-data-pass-through), but this comes with a lot of tradeoffs and has a code smell to me.

There is a [`sphinx-svg2pdfconverter`](https://pypi.org/project/sphinxcontrib-svg2pdfconverter/), but I also don't want to add another dependency.

SecondSkoll replied:
I previously found it best to use `.. only::` directives combined with `.. raw:: latex` directives to include SVG images with [specific LaTeX support](https://www.ctan.org/pkg/svg), but it is a real pain. I previously used the SVG converter but found there were times that it made dodgy edges and rasterised things poorly. It's better (IMO) to know what the image looks like when you include it and avoid the processing at a stage where you can't really do anything about how it handles the image.

SecondSkoll committed:
Update custom_conf.py

remove proprietary notice

SecondSkoll committed:
Update Makefile

Add PDF generation commands to make help

dviererbe commented on docs/Makefile:
```diff
@@ -28,6 +28,8 @@ help:
         "* check spelling: make spelling \n" \
         "* check inclusive language: make woke \n" \
         "* check accessibility: make pa11y \n" \
+	"* build PDF: make pdf \n" \
+	"* force install packages for PDF generation: make pdf-prep-force \n" \
         "* other possible targets: make <press TAB twice> \n" \
```
nitpick: something is wrong with the indentation here:
```suggestion
        "* check accessibility: make pa11y \n" \
        "* build PDF: make pdf \n" \
        "* force install packages for PDF generation: make pdf-prep-force \n" \
        "* other possible targets: make <press TAB twice> \n" \
```

dviererbe commented on docs/Makefile:
```diff
@@ -99,6 +101,35 @@ woke: woke-install
+
+pdf-prep-force:
+	@. $(VENV); apt-get update
+	@. $(VENV); apt-get upgrade -y
```
nitpick: `apt-get upgrade -y` is not strictly necessary and should be the choice of the user
```suggestion
```

dviererbe commented on docs/Makefile:
```diff
@@ -99,6 +101,35 @@ woke: woke-install
+# Entrypoint to install system packages, separate from usual workflow due to permanence
+
+pdf-prep-force:
+	@. $(VENV); apt-get update
+	@. $(VENV); apt-get upgrade -y
+	@. $(VENV); apt-get install --no-install-recommends -y latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex plantuml xindy tex-gyre dvipng \
```
nitpick: I think it is useful for the user to see that the apt commands were executed if something fails.

nitpick: the `\` at the end is not needed

issue: This will probably fail without `sudo`

```suggestion
	. $(VENV); apt-get update
	. $(VENV); apt-get upgrade -y
	. $(VENV); apt-get install --no-install-recommends -y latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex plantuml xindy tex-gyre dvipng
```

dviererbe replied:
note: also the `. $(VENV);` is not needed as the venv only isolates the python package installs and not apt installs

dviererbe reviewed (changes requested):
I do not claim to understand all the latex changes you did, but they look sound to me and I verified that `make pdf` works. Thanks a lot! ❤️ 

There are some nitpicks. I will merge as soon as they are resolved.

dviererbe commented:
> Apart from one little nit, LGTM :) it'll be nice to have the local PDF build ability as well - although we might want to also add a little documentation to explain how to use it (or add it to the make "help" list so when users run `make` they can see it as an option)

We could add this as an issue for the Open Documentation Academy

CC: @rkratky

SecondSkoll committed:
Update docs/Makefile

Accept suggestion regarding optional upgrade

Co-authored-by: Dominik Viererbe <github@dviererbe.de>

SecondSkoll committed:
Update docs/Makefile

Fix tabs/spaces issue

Co-authored-by: Dominik Viererbe <github@dviererbe.de>

SecondSkoll committed:
Update Makefile

Remove silenced package installs, remove unnecessary VENV activation from package installs
